domino/domino.py

```diff
@@ -122,14 +122,30 @@ def runs_start_blocking(self, command, isDirect=False, commitId=None,
                         Maximum number of seconds to wait for a task to
                         complete. If this threshold is exceeded, an exception
                         is raised.
+
+        retry_count : int (Optional)
+                        Maximum number of retry to do while polling
+                        (in-case of transient httpd errors). If this
+                        threshold exceeds, an exception is raised.
         """
         run_response = self.runs_start(command, isDirect, commitId, title,
                                        tier, publishApiEndpoint)
         run_id = run_response['runId']
 
         poll_start = time.time()
+        current_retry_count = 0
         while True:
-            run_info = self.get_run_info(run_id)
+            try:
+                run_info = self.get_run_info(run_id)
+            except requests.exceptions.RequestException as e:
+                current_retry_count += 1
+                self._logger.warn(f'Failed to get run info for runId: {run_id} : {e}')
+                if current_retry_count > retry_count:
+                    raise Exception(f'Cannot get run info, max retry {retry_count} exceeded') from None
+                else:
+                    self._logger.info(f'Retrying ({current_retry_count}/{retry_count}) getting run info ...')
+                    continue
```

Comment:
Do we want to go to sleep before retrying?

Reply:
Will add a sleep about `poll_freq ` here